pkg/otlp/metrics/metrics_remapping.go

```diff
@@ -70,7 +70,9 @@ func remapSystemMetrics(all pmetric.MetricSlice, m pmetric.Metric) {
 		copyMetric(all, m, "system.disk.in_use", 1)
 	}
 	// process.* and system.* metrics need to be prepended with the otel.* namespace
-	m.SetName("otel." + m.Name())
+	newm := all.AppendEmpty()
```

Comment:
Chatted with @liustanley , only the metrics from the previous `extractSystemMetrics` were kept: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23445/files#diff-59a3a2cae4395f15cad4800806f9dc7680aed0f303dd896c0ca8663e0fd0ac35L56

For `process.*` metrics, only the ones with `otel.*` prefix should be exported, original ones are expected to be overriden.

Reply:
Sorry but I don't see the difference between this implementation and the one in the linked code. All metrics that start with system.* and process.* seem to get the prefix. I don't see any special case for `process.*`

Reply:
Both of system.* and process.* metrics get the otel. prefix - that's right. In addition to that, the original system.* metrics (no otel. prefix) should be preserved / copied, while process.* metrics  don't.

Reply:
I understand, but I don't see that in the old code. The behaviour seems the same for both process and system

Reply:
Either way, in my latest commit, I think that I have applied what you meant. PTAL